src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -308,6 +313,21 @@ def _slice(
         assert common.is_relative_index_sequence(slice_)
         return new_domain, slice_
 
+    def _refactor_slice(
+        self, idx: slice
+    ) -> tuple[common.Dimension, common.UnitRange] | common.AnyIndexSpec:
+        if isinstance(idx.start, tuple) or isinstance(idx.stop, tuple):
+            is_start_none: bool = idx.start is None
+            is_stop_none: bool = idx.stop is None
+            dim_idx = list(self.domain.dims).index(idx.stop[0] if is_start_none else idx.start[0])
+            if not is_start_none and not is_stop_none:
+                # assert dimension in upper and lower bounds are the same
+                assert idx.start[0] == idx.stop[0]
```

Comment:
This should not be an assert, because it is a user-facing issue. Should be a proper error message.